### DISCUSSION TITLE
Request: Add workflow name in event check_run payload

### DISCUSSION THREAD

**Discussion by missedone** [community/community#25919] [Actions]
<p>hi folks<br>
i’m trying to develop a solution to auto scale out self-hosted runner.<br>
i was inspired this article: <a href="https://040code.github.io/2020/05/25/scaling-selfhosted-action-runners" rel="nofollow noopener">https://040code.github.io/2020/05/25/scaling-selfhosted-action-runners</a><br>
basically, it listens on event check_run to trigger auto scale out.<br>
however, in my use cases, one github organization may have multiple self-hosted runners with different labels.<br>
with current check_run event payload, i can’t tell what runner to be scaled:</p>
<pre><code class="lang-auto">{
  "action": "created",
  "check_run": {
    "id": 936829714,
    ...
    "status": "queued",
    ...
    "name": "Build",
    "check_suite": {
        ...
    }
    ...
  }
}
</code></pre>
<p>I think with workflow name, i can get the runner label than knows which runner to scale. but I can only see name “Build” which is job name and not unique.</p>
<p>could you add “workflow” name in the event payload? or suggest a way I can tell which runner to scale.<br>
thanks</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/missedone">@missedone</a>,</p>
<p>The workflow name is included in GITHUB CONTEXT: <code>${{ github.workflow }}</code>, however it’s not included in the check_run event payload. You can raise a feature_request <a href="https://support.github.com/contact/feedback?contact%5Bcategory%5D=actions" rel="nofollow noopener">here</a> where github product manager will take a review.</p>
<p>Since you can get the check_run id in the payload,  you can get the workflow file with below link:<br>
<a href="https://github.com/%7Borg%7D/%7Brepo%7D/actions/runs/%7Bcheckrun_id%7D/workflow" rel="nofollow noopener">https://github.com/{org}/{repo}/actions/runs/{checkrun_id}/workflow</a>, parse the selfhosted runner info then.</p>
<p>Thanks</p>

**Comment by missedone** [comment]
<p>thanks, sounds a good workaround for now.<br>
also i raised the feature request.</p>
